src/licensedcode/detection.py

```diff
@@ -638,6 +632,23 @@ def dict_fields(attr, value):
         return attr.asdict(self, filter=dict_fields)
 
 
+def get_detections_by_id(license_detections):
+    """
+    Get a dict(hashmap) where each item is: {detection.identifier: all_detections} where
+    `all_detections` is all detections in `license_detections` whose detection.identifier
+    is the same.
+    """
+    detections_by_id = {}
```

Comment:
This would be simpler:
```suggestion
    detections_by_id = defaultdict(list)
    for detection in license_detections:
        detections_by_id[detection.identifier].append(detection)
```

Reply:
Thanks! Added this.